draft-ietf-mls-protocol.md

```diff
@@ -2752,7 +2758,9 @@ welcome_key = KDF.Expand(welcome_secret, "key", key_length)
     children are non-empty and have the hash of this node set as their
     `parent_hash` value (if the child is another parent) or has a `parent_hash`
     extension in the KeyPackage containing the same value (if the child is a
-    leaf).
+    leaf). If either of the node's children is empty, and in particular does not
+    have a parent hash, then its respective children's `parent_hash` values have
+    to be considered instead.
```

Comment:
This seems incorrect to me.  The  value of `parent_hash` is set from the direct parent, so it's useless to look at the blank node's children.  You don't have enough information to verify that `parent_hash`; you're missing what was in the blank node.  You also can't reset the `parent_hash` to "skip over" the blank node when you set it to blank, because that would invalidate `parent_hash` values below it in the tree and ultimately, the signature on the leaf.